edx/edx-enterprise
# ENT-696 update course enrollment view for enterprise enabled course modes

@asadiqbal08 @saleem-latif @douglashall 

**Description:** update course enrollment view for enabled course mode in enterprise catalogs

**JIRA:** [ENT-696](https://openedx.atlassian.net/browse/ENT-696)

**Dependencies:** N/A

**Merge deadline:** 20 Oct, 2017

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-696-update-course-enrollment-for-catalog-queryparam` in `edx-platform`.

**Testing instructions:**

1. Create enterprise with a catalog
2. Add a record in table `Enterprise Customer Catalogs` against the above enterprise.
3. Provide desired course mode (in order) to display on enterprise course enrollment page
<img width="973" alt="screen shot 2017-10-13 at 6 07 46 pm" src="https://user-images.githubusercontent.com/5072991/31547815-25cac938-b042-11e7-9e99-ca90fb70359f.png">

4. Now go to enterprise course enrollment page as a learner and verify that the displayed course modes are from the list provided in table `Enterprise Customer Catalogs` against the enterprise.
5. Now update the value for `Enabled course modes:` in enterprise customer catalog record and provide course modes which don't exist for the course.
6.  Go to enterprise course enrollment page as a learner and verify that learner sees a generic info message `Something happened. This course is not available. Please start over and select a different course.` and there are no course modes listed on the page and also there is no `Continue` button.
<img width="829" alt="screen shot 2017-10-13 at 6 39 46 pm" src="https://user-images.githubusercontent.com/5072991/31549129-a5f8cf98-b046-11e7-8361-410afe045e2f.png">


**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
